port/stmicro/stm32/src/hals/STM32F103/clocks/ClockNode.zig

```diff
@@ -0,0 +1,449 @@
+const std = @import("std");
+const comptimePrint = std.fmt.comptimePrint;
+
+pub const Limit = struct {
+    min: ?f32 = null,
+    max: ?f32 = null,
+    min_expr: ?[]const u8 = null,
+    max_expr: ?[]const u8 = null,
+};
+
+pub const ClockNodesTypes = enum {
+    source,
+    multi,
+    mul,
+    mulfrac,
+    div,
+    output,
```

Comment:
About naming, If we have output should we have input instead of source? Those multi means in_and_out?